langchain/vectorstores/milvus.py

```diff
@@ -11,419 +12,837 @@
+        metric_type: str = index["index_param"]["metric_type"]
+        self.search_params = self.default_search_params[index_type]
+        self.search_params["metric_type"] = metric_type
 
     def add_texts(
         self,
         texts: Iterable[str],
         metadatas: Optional[List[dict]] = None,
-        partition_name: Optional[str] = None,
         timeout: Optional[int] = None,
+        batch_size: int = 1000,
         **kwargs: Any,
     ) -> List[str]:
         """Insert text data into Milvus.
 
-        When using add_texts() it is assumed that a collecton has already
-        been made and indexed. If metadata is included, it is assumed that
-        it is ordered correctly to match the schema provided to the Collection
-        and that the embedding vector is the first schema field.
+        Inserting data when the collection has not be made yet will result 
+        in creating a new Collection. The data of the first entity decides
+        the schema of the new collection, the dim is extracted from the first
+        embedding and the columns are decided by the first metadata dict.
+        Metada keys will need to be present for all inserted values. At
+        the moment there is no None equivalent in Milvus.
 
         Args:
-            texts (Iterable[str]): The text being embedded and inserted.
-            metadatas (Optional[List[dict]], optional): The metadata that
-                corresponds to each insert. Defaults to None.
-            partition_name (str, optional): The partition of the collection
-                to insert data into. Defaults to None.
-            timeout: specified timeout.
+            texts (Iterable[str]): The texts to embed, it is assumed 
+                that they all fit in memory.
+            metadatas (Optional[List[dict]]): Metadata dicts attached to each of 
+                the texts. Defaults to None.
+            timeout (Optional[int]): Timeout for each batch insert. Defaults 
+                to None.
+            batch_size (int, optional): Batch size to use for insertion. 
+                Defaults to 1000.
+
+        Raises:
+            MilvusException: Failure to add texts
 
         Returns:
             List[str]: The resulting keys for each inserted element.
         """
-        insert_dict: Any = {self.text_field: list(texts)}
+        from pymilvus import MilvusException
+
+        texts = list(texts)
         try:
-            insert_dict[self.vector_field] = self.embedding_func.embed_documents(
-                list(texts)
+            embeddings = self.embedding_func.embed_documents(
+                texts
             )
         except NotImplementedError:
-            insert_dict[self.vector_field] = [
+            embeddings = [
                 self.embedding_func.embed_query(x) for x in texts
             ]
+        # If the collection hasnt been initialized yet, perform all steps to do so
+        if self.col is None:
```

Comment:
My preference here is to create a `self._initialize` function which performs all of these substeps since these are called both here and in `__init__()`.